cpp/src/arrow/adapters/orc/adapter_util.cc

```diff
@@ -207,7 +207,13 @@ Status AppendTimestampBatch(liborc::ColumnVectorBatch* column_vector_batch,
   const int64_t* nanos = batch->nanoseconds.data() + offset;
 
   auto transform_timestamp = [seconds, nanos](int64_t index) {
-    return seconds[index] * kOneSecondNanos + nanos[index];
+    // Firebolt Start
+    // Convert ORC's timestamp column with a resolution of nanoseconds to Arrow's
+    // timestamp column with a resolution of microseconds by truncating to microseconds.
+    const int64_t seconds_us = seconds[index] * kOneSecondMicros;
+    const int64_t subseconds_us = nanos[index] / kOneMicroNanos;
+    return seconds_us + subseconds_us;
```

Comment:
can this overflow?

Reply:
Yes it can. Should I do checked multiplication and addition?

Reply:
Switched to checked multiplication